test/integration/spec/connect.js

```diff
@@ -720,6 +721,52 @@ describe('connect', function() {
     });
   });
 
+  describe.only('called with a Room name and', () => {
+    let sid;
+    let cancelablePromise;
+
+    before(async () => {
+      sid = await createRoom(randomName(), defaults.topology);
+      const options = Object.assign({ name: sid, tracks: [] }, defaults);
+      cancelablePromise = connect(getToken(randomName()), options);
+    });
+
+    after(() => completeRoom(sid));
+
+    it('should return a promise', async () => {
+      const onFinally = sinon.stub();
+      await cancelablePromise.finally(onFinally);
+      sinon.assert.calledOnce(onFinally);
+    });
+
+    it('should resolve with a room and finally is called', async () => {
+      const onFinally = sinon.stub();
+      let room;
+      let errorThrown = null;
+      try {
+        room = await cancelablePromise;
+      } catch (error) {
+        errorThrown = error;
+      }
+      await cancelablePromise.finally(onFinally);
+      sinon.assert.calledOnce(onFinally);
+      assert(!errorThrown);
+      assert(room instanceof Room);
+    });
+
+    it('should reject and finally is called', async () => {
+      const onFinally = sinon.stub();
+      try {
+        await cancelablePromise;
+        throw new Error('Connecting to room, but expecting to cancel');
+      } catch (error) {
+        assert(error);
```

Comment:
I don't think we should perform assertions inside a catch block. There are chances that catch block is not executed, skipping your assertion. You can do the same pattern you did on the previous test above instead.